### PR TITLE
Add erc20 template tests

### LINKED ISSUE #576: Implement ERC20 Template unit tests for V4
Implement ERC20 Template unit tests for V4 from [here](https://github.com/oceanprotocol/contracts/blob/v4main/test/unit/datatokens/ERC20Template.test.js).

### PR DESCRIPTION
**Pull request by maldo-ctrl**
Fixes #576 .

Changes proposed in this PR:

- Add ERC20 Template unit tests
- Add  `erc20_token.py` roles enum
- Add  `erc20_token.py` missing functions and fixed wrong permissions return type
- Removed duplicated/non-existent functions in `erc20_token.py`

Screenshot from 2021-11-11 11-58-38

image

### CODE REVIEW

**Review by mariacarmina — COMMENTED** [review]
Can you post here a screenshot of the coverage please? In rest, it looks fine from my side.

**Comment by maldo-ctrl** [comment]
> Can you post here a screenshot of the coverage please? In rest, it looks fine from my side.

image

File: ocean_lib/models/v4/erc20_token.py [hunk 1]
```diff
@@ -248,9 +256,6 @@ def cap(self) -> int:
     def is_initialized(self) -> bool:
         return self.contract.caller.isInitialized()
 
-    def calculate_fee(self, amount: int, fee_percentage: int) -> int:
-        return self.contract.caller.calculateFee(amount, fee_percentage)
-
```
Comment (maldo-ctrl) [inline ocean_lib/models/v4/erc20_token.py:253]: This function doesn't exist in Erc20 Template

File: ocean_lib/models/v4/erc20_token.py [hunk 2]
```diff
@@ -227,9 +238,6 @@ def set_publishing_market_fee(
             from_wallet,
         )
 
-    def permissions(self, account: str) -> tuple:
```
Comment (maldo-ctrl) [inline ocean_lib/models/v4/erc20_token.py:230]: Removing duplicated function https://github.com/oceanprotocol/ocean.py/blob/e8dea08044e31436cddb1a106f07c5252986a3ba/ocean_lib/models/v4/erc20_token.py#L294

File: tests/resources/helper_functions.py [hunk 3]
```diff
@@ -222,7 +222,6 @@ def deploy_erc721_erc20(
     """Helper function to deploy an ERC721Token using erc721_publisher Wallet
     and an ERC20Token data token with the newly ERC721Token using erc20_minter Wallet
     if the wallet is provided.
-    :rtype: ERC721Token or (ERC721Token, ERC20Token)
```
Comment (maldo-ctrl) [inline tests/resources/helper_functions.py:225]: This dock strings `rtype` param was unnecessary and causing problems with IDEs. This function already returns: `Union[ERC721Token, Tuple[ERC721Token, ERC20Token]]`

Comment (mariacarmina) [reply tests/resources/helper_functions.py:225]: We can update it.

File: ocean_lib/models/v4/test/test_erc20_token.py [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+        erc20_minter=publisher_wallet,
+        cap=web3.toWei(publishMarketFeeAmount, "ether"),
+    )
+
+    # Check erc20 params
+    assert erc20.get_id() == 1
+    assert erc20.contract.caller.name() == "ERC20DT1"
+    assert erc20.symbol() == "ERC20DT1Symbol"
+    assert erc20.decimals() == 18
+    assert erc20.cap() == web3.toWei(publishMarketFeeAmount, "ether")
```
Comment (mariacarmina) [inline ocean_lib/models/v4/test/test_erc20_token.py:134]: can we replace with snake case or put the actual value here if it's used one time?

**Review by mariacarmina — APPROVED** [review]

**Pull request merged**